src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs

```diff
@@ -0,0 +1,169 @@
+    /// </summary>
+    /// <example>
+    /// Use in application host
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var milvus = builder.AddMilvus("milvus");
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///                  .WithReference(milvus);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// This version the package defaults to the 2.3-latest tag of the milvusdb/milvus container image.
+    /// The .NET client library uses the gRPC port by default to communicate and this resource exposes that endpoint.
+    /// A web-based administration tool for Milvus can also be added using <see cref="WithAttu"/>.
+    /// </remarks>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency</param>
+    /// <param name="apiKey">The parameter used to provide the auth key/token user for the Milvus resource.</param>
+    /// <param name="grpcPort">The host port of gRPC endpoint of Milvus database.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MilvusServerResource}"/>.</returns>
+    public static IResourceBuilder<MilvusServerResource> AddMilvus(this IDistributedApplicationBuilder builder,
+        string name,
+        IResourceBuilder<ParameterResource> apiKey,
+        int? grpcPort = null)
+    {
+        ArgumentNullException.ThrowIfNull(apiKey, nameof(apiKey));
+
+        var tokenParameter = apiKey.Resource;
+        var milvus = new MilvusServerResource(name, tokenParameter);
+
+        return builder.AddResource(milvus)
+            .WithImage(MilvusContainerImageTags.Image, MilvusContainerImageTags.Tag)
+            .WithImageRegistry(MilvusContainerImageTags.Registry)
+            .WithHttpEndpoint(port: grpcPort, targetPort: MilvusPortGrpc, name: MilvusServerResource.PrimaryEndpointName)
+            .WithEndpoint(MilvusServerResource.PrimaryEndpointName, endpoint =>
+            {
+                endpoint.Transport = "http2";
+            })
+            .WithEnvironment("COMMON_STORAGETYPE", "local")
+            .WithEnvironment("ETCD_USE_EMBED", "true")
+            .WithEnvironment("ETCD_DATA_DIR", "/var/lib/milvus/etcd")
+            .WithEnvironment("COMMON_SECURITY_AUTHORIZATIONENABLED", "true")
+            .WithArgs("milvus", "run", "standalone");
+    }
+
+    /// <summary>
+    /// Adds a Milvus database to the application model.
+    /// </summary>
+    /// <example>
+    /// Use in application host
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var booksdb = builder.AddMilvus("milvus");
+    ///   .AddDatabase("booksdb");
+    /// 
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///                  .WithReference(booksdb);
```

Comment:
(nit) - it would be nice to use the same whitespace pattern in an example.

```suggestion
    /// var booksdb = builder.AddMilvus("milvus");
    ///   .AddDatabase("booksdb");
    /// 
    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
    ///   .WithReference(booksdb);
```